pkg/pillar/cmd/zedagent/zedagent.go

```diff
@@ -1786,42 +1846,57 @@ func triggerPublishHwInfo(ctxPtr *zedagentContext) {
 	}
 }
 
-func triggerPublishDevInfo(ctxPtr *zedagentContext) {
+func triggerPublishHwInfo(ctxPtr *zedagentContext) {
+	triggerPublishHwInfoWithDest(ctxPtr, AllDest)
+}
+
+func triggerPublishDevInfoWithDest(ctxPtr *zedagentContext, dest infoDest) {
 
 	log.Function("Triggered PublishDeviceInfo")
 	select {
-	case ctxPtr.TriggerDeviceInfo <- struct{}{}:
+	case ctxPtr.TriggerDeviceInfo <- dest:
 		// Do nothing more
 	default:
 		// This occurs if we are already trying to send a device info
 		// and we get a second and third trigger before that is complete.
 		log.Warnf("Failed to send on PublishDeviceInfo")
 	}
-	triggerLocalDevInfoPOST(ctxPtr.getconfigCtx)
+	if dest&LocalServerDest != 0 {
+		triggerLocalDevInfoPOST(ctxPtr.getconfigCtx)
+	}
 }
 
-func triggerPublishLocationToController(ctxPtr *zedagentContext) {
+func triggerPublishDevInfo(ctxPtr *zedagentContext) {
+	triggerPublishDevInfoWithDest(ctxPtr, AllDest)
+}
+
+func triggerPublishLocationToControllerWithDest(ctxPtr *zedagentContext, dest infoDest) {
 	if ctxPtr.getconfigCtx.locationCloudTickerHandle == nil {
 		// Location reporting task is not yet running.
 		return
 	}
 	log.Function("Triggered publishLocationToController")
-	flextimer.TickNow(ctxPtr.getconfigCtx.locationCloudTickerHandle)
+	ctxPtr.TriggerCloudLocationInfo <- dest
 }
 
-func triggerPublishAllInfo(ctxPtr *zedagentContext) {
+func triggerPublishLocationToController(ctxPtr *zedagentContext) {
```

Comment:
Shouldn't this be `triggerPublishLocation` (without `ToController`)?
Also `triggerPublishLocationToControllerWithDest` should be probably just `triggerPublishLocationWithDest`

Reply:
Yep, can be removed.